openapi.yaml

```diff
@@ -0,0 +1,2411 @@
+        This can be combined with startd_ate_lte parameter to receive only the selected period.
+    FilterStartDateLTE:
+      in: query
+      name: start_date_lte
+      schema:
+        type: string
+        format: date-time
+      required: false
+      description: >
+        Returns objects less or equal the specified date.
+
+        This can be combined with start_date_gte parameter to receive only the selected period.
+    FilterEndDateGTE:
+      in: query
+      name: end_date_gte
+      schema:
+        type: string
+        format: date-time
+      required: false
+      description: >
+        Returns objects greater or equal the specified date.
+
+        This can be combined with start_date_lte parameter to receive only the selected period.
+    FilterEndDateLTE:
+      in: query
+      name: end_date_lte
+      schema:
+        type: string
+        format: date-time
+      required: false
+      description: >
+        Returns objects less than or equal to the specified date.
+
+        This can be combined with start_date_gte parameter to receive only the selected period.
+    FilterDurationGTE:
+      in: query
+      name: duration_gte
+      schema:
+        type: number
+      required: false
+      description: >
+        Returns objects greater than or equal to the specified values.
+
+        This can be combined with duration_lte parameter to receive only the selected period.
+    FilterDurationLTE:
+      in: query
+      name: duration_lte
+      schema:
+        type: number
+      required: false
+      description: >
+        Returns objects less than or equal to the specified values.
+
+        This can be combined with duration_gte parameter to receive only the selected range.
+    FilterState:
+      in: query
+      name: state
+      schema:
+        type: array
+        items:
+          type: string
+      required: false
+      description:
+        The value can be repeated to retrieve multiple matching values (OR condition).
+    FilterPool:
+      in: query
+      name: pool
+      schema:
+        type: array
+        items:
+          type: string
+      required: false
+      description:
+        The value can be repeated to retrieve multiple matching values (OR condition).
+    FilterQueue:
+      in: query
+      name: Queue
+      schema:
+        type: array
+        items:
+          type: string
+      description:
+        The value can be repeated to retrieve multiple matching values (OR condition).
+
+    # Other parameters
+    FileToken:
+      in: path
+      name: file_token
+      schema:
+        type: string
+      required: true
+      description: >
+        The key containing the encrypted path to the file. Encryption and decryption take place only on
+        the server. This prevents the client from reading an non-DAG file. This also ensures API
+        extensibility, because the format of encrypted data may change.
+
+    UpdateMask:
+      in: query
+      name: update_mask
```

Comment:
Why is it called "UpdateMask" - Just curious, no comment here

Reply:
Mostly just copying https://google.aip.dev/134#patch-and-put

Reply:
Yes. I tried to stick to one recommendation on building an API. We don't have to come up with a new idea if we can rely on tested solutions.